### PR TITLE
documentation for workspaces

### PR DESCRIPTION
**Pull request by memsharded**
Docs for https://github.com/conan-io/conan/pull/4481

### CODE REVIEW

File: developing_packages/workspaces.rst [hunk 1]
```diff
@@ -76,88 +184,125 @@ Note that nothing will really be installed in the local cache, all the dependenc
...
+    [build_folder]
+    build
+
+
+Notes
+-----
 
-Known limitations
------------------
+Not that this way of developing packages shouldn't be used to create the final packages (you could try to use ``conan export-pkg``), but instead,
```
Comment (danimtb) [inline developing_packages/workspaces.rst]:
```suggestion
Note that this way of developing packages shouldn't be used to create the final packages (you could try to use :command:`conan export-pkg`). Instead,
```

File: developing_packages/workspaces.rst [hunk 2]
```diff
@@ -76,88 +184,125 @@ Note that nothing will really be installed in the local cache, all the dependenc
...
+a full package creation with ``conan create`` (best in CI) is recommended.
```
Comment (danimtb) [inline developing_packages/workspaces.rst]:
```suggestion
a full package creation with :command:`conan create` is recommended. 
```

File: reference/commands/development/workspace.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+.. _conan_workspace:
+
+conan workspace
+===============
+
+.. code-block:: bash
+
+    $ conan workspace [-h] {install} ...
+
+command to manage workspaces
```
Comment (danimtb) [inline reference/commands/development/workspace.rst]:
```suggestion
Command to manage workspaces.
```

File: developing_packages/workspaces.rst [hunk 4]
```diff
@@ -14,60 +14,168 @@ The local development flow can help, but it still requires using :command:`expor
...
+    set(PACKAGE_chat_SRC "<path>/examples/workspace/cmake/chat/src")
+    set(PACKAGE_chat_BUILD "<path>/examples/workspace/cmake/chat/build/Debug")
+
+    macro(conan_workspace_subdirectories)
+        add_subdirectory(${PACKAGE_say_SRC} ${PACKAGE_say_BUILD})
+        add_subdirectory(${PACKAGE_hello_SRC} ${PACKAGE_hello_BUILD})
+        add_subdirectory(${PACKAGE_chat_SRC} ${PACKAGE_chat_BUILD})
+    endmacro()
+
+This file can be included in your root *CMakeLists.txt* (this one is not generated, you provide it, so you can customize it
```
Comment (danimtb) [inline developing_packages/workspaces.rst]: I would rephrase this two lines to something like:

> The consumption of the packages is done including this file in an user-defined *CMakeLists.txt* file so you can customize it as much as you want. Here you can see a simple one included in this project:

File: developing_packages/workspaces.rst [hunk 5]
```diff
@@ -76,88 +184,125 @@ Note that nothing will really be installed in the local cache, all the dependenc
...
+    conan_basic_setup(NO_OUTPUT_DIRS)
+
+    add_library(hello hello.cpp)
+    conan_target_link_libraries(hello)
+
+
+The last ``conan_target_link_libraries(hello)`` is a helper that does the right linking with Debug/Release libraries (also works when using cmake
+targets).
+
+Lets use this. Note it is very important to install both Debug and Release configurations straight ahead, if we want to later switch between them
```
Comment (danimtb) [inline developing_packages/workspaces.rst]:
```suggestion
Make sure you install both Debug and Release configurations straight ahead, so we can switch build type later
```

File: developing_packages/workspaces.rst [hunk 6]
```diff
@@ -76,88 +184,125 @@ Note that nothing will really be installed in the local cache, all the dependenc
...
 
-    $ cd build
-    $ cmake .. -G "Visual Studio 14 Win64" # Adapt accordingly to your Conan profile
-    # Now build it. You can also open your IDE and build
-    $ cmake --build . --config Release
-    $ ./A/Release/app.exe
-    Hello World C Release!
-    Hello World B Release!
-    Hello World A Release!
+    This is because the default ``conan_basic_setup()`` does define output directories for artifacts as *bin*, *lib*, etc, which is
```
Comment (jgsogo) [inline developing_packages/workspaces.rst]: Why not leaving the standard `conan_basic_setup()` call and then make the `layout` file match to it? Explanation, if needed, will be closer to the layout file description, not to the CMake one.

File: reference/commands/development/workspace.rst [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    -pr PROFILE, --profile PROFILE
+                            Apply the specified profile to the install command
+    -r REMOTE, --remote REMOTE
+                            Look in the specified remote server
+    -s SETTINGS, --settings SETTINGS
+                            Settings to build the package, overwriting the
+                            defaults. e.g., -s compiler=gcc
+    -u, --update          Check updates exist from upstream remotes
+
+    Opens the package ``<reference>`` in editable mode in the user folder ``<path>``
```
Comment (jgsogo) [inline reference/commands/development/workspace.rst]: Remove, these lines doesn't correspond to `conan workspaces install ...` command docs.

File: reference/commands/development/workspace.rst [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    -o OPTIONS, --options OPTIONS
+                            Define options values, e.g., -o Pkg:with_qt=true
+    -pr PROFILE, --profile PROFILE
+                            Apply the specified profile to the install command
+    -r REMOTE, --remote REMOTE
+                            Look in the specified remote server
+    -s SETTINGS, --settings SETTINGS
+                            Settings to build the package, overwriting the
+                            defaults. e.g., -s compiler=gcc
+    -u, --update          Check updates exist from upstream remotes
```
Comment (jgsogo) [inline reference/commands/development/workspace.rst]: I would add a note saying that some of these options don't apply to packages referenced in the _workspace definition file_, I suppose that those ones are not being built or looked for in the remotes... those packages must be present in advance, no?

File: reference/config_files/editable_layout.rst [hunk 9]
```diff
@@ -33,9 +40,13 @@ This file can live in the conan cache, in the ``.conan/layouts`` folder, or in a
 inside the source repo.
 
 If there exists a ``.conan/layouts/default`` layout file in the cache and no layout file is specified
-in the ``conan link <path> <reference>`` command, that file will be used.
+in the ``conan editable add <path> <reference>`` command, that file will be used.
```
Comment (jgsogo) [inline reference/config_files/editable_layout.rst]: That file (`.conan/layouts/default`) will no longer be used by default

**Comment by danimtb** [comment]
please @memsharded, address the review above to have this merged. The PR with the code changes was already merged and this one is pending

File: reference/config_files/editable_layout.rst (hunk 9 above)
Comment (memsharded) [reply reference/config_files/editable_layout.rst]: Yes, that has not changed. If you don't specify a layout, that file will be used by default.

File: developing_packages/workspaces.rst (hunk 6 above)
Comment (memsharded) [reply developing_packages/workspaces.rst]: Good point, I'll try to improve it later, but I am going to submit my review, so it can be merged soon.
